### ISSUE TITLE
Determine `greater_is_better` from eval_metric? (Callbacks)

### ISSUE THREAD

**Issue by Innixma** [catboost/catboost#1890]
Problem: How to determine `greater_is_better` from eval_metric?
catboost version: v1.0

Thanks to the CatBoost team for adding callback support, it has been very useful! (#1304)

Now that callbacks are supported in CatBoost, I want to be able to create my own custom early stopping method, such as using a dynamic early stopping rounds value depending on how many iterations have already been trained, or by leveraging the actual scores and the magnitude of their differences. However, it is unclear how to determine if one score is better than another.

For example, in LightGBM callbacks:

```
def _callback(env):
    eval_result = env.evaluation_result_list[0]
    _, eval_metric, cur_score, greater_is_better = eval_result
```

For each iteration this is provided which is enough to compare and see if iteration 2 is better than iteration 1 via:

```
if greater_is_better:
    is_better = cur_score > self.best_score
else:
    is_better = cur_score < self.best_score
```

I don't see how I can get the `greater_is_better` value in CatBoost and I greatly want to avoid hard-coding on my end the value for every possible metric (including custom metrics).

Code example:

```


import numpy as np
from catboost import CatBoostClassifier, Pool

np.random.seed(0)
train_data = np.random.randint(0, 100, size=(1000, 5))
train_labels = np.random.randint(0, 2, size=1000)

val_data = np.random.randint(0, 100, size=(1000, 5))
val_labels = np.random.randint(0, 2, size=1000)


class CustomEarlyStopCallback:
    def __init__(self, eval_metric, greater_is_better, compare_key='validation'):
        self.compare_key = compare_key
        self.eval_metric = eval_metric
        self.greater_is_better = greater_is_better
        self.best_score = None
        self.best_iter = None

    def after_iteration(self, info):
        cur_score = info.metrics[self.compare_key][self.eval_metric][-1]
        if self.best_score is None:
            is_better = True
        else:
            if self.greater_is_better:
                is_better = cur_score > self.best_score
            else:
                is_better = cur_score < self.best_score
        if is_better:
            self.best_score = cur_score
            self.best_iter = info.iteration
        iter_since_best = info.iteration - self.best_iter
        print(f'Cur Iter: {info.iteration}, Cur Score: {cur_score}, Best Iter: {self.best_iter}, Best Score: {self.best_score}, Iter since Best: {iter_since_best}')

        return True


# eval_metric = 'Logloss'  # --> greater_is_better=False
eval_metric = 'Accuracy'  # --> greater_is_better=True
greater_is_better = True  # FIXME: How do I know what the correct value is?
callbacks = [CustomEarlyStopCallback(eval_metric=eval_metric, greater_is_better=greater_is_better)]

eval_set = Pool(data=val_data, label=val_labels)
model = CatBoostClassifier(iterations=10, loss_function='Logloss', eval_metric=eval_metric)
model.fit(train_data, train_labels, eval_set=eval_set, callbacks=callbacks)

```

Ideal solution:

```
from catboost import is_metric_greater_is_better
assert is_metric_greater_is_better('Logloss') == False
assert is_metric_greater_is_better('Accuracy') == True
assert is_metric_greater_is_better(custom_metric) == custom_metric.is_max_optimal()
```

**Comment by andrey-khropov** [comment]
Deployed in release 1.0.3.
